ESQL: Add version to docs (#107225)
elastic/elasticsearch: docs/reference/esql/esql-version.asciidoc

[[esql-version]]
=== {esql} language versions

++++
<titleabbrev>Language versions</titleabbrev>
++++

[discrete]
[[esql-versions-released]]
==== Released versions

> {esql} is released at version `2024.04.01.🚀`.

Reviewer: ```suggestion
* Version `2024.04.01.🚀`.
```

Sets up the bulleted list for additions in future

> This is the only available version for now.

Reviewer: ```suggestion
```

nit: might be self-evident

[discrete]
[[esql-versions-explanation]]
=== How versions work

> {esql} versions are separate from Elasticsearch versions.

Reviewer: ```suggestion
{esql} language versions are independent of {es} versions.
```

> Their goal is to make sure that queries that were once written
> remain valid even as Elasticsearch and {esql} evolve.

Reviewer: ```suggestion
Versioning the language ensures that your queries will always
remain valid for that version, independent of new {es} and {esql} releases.
```

nit: rewording

> For instance, the following query

Reviewer: It doesn't read well to have a code sample mid-sentence, so I'd recommend moving lines 41 + 42 up here.

[source,console]
----
POST /_query?format=txt
{
  "version": "2024.04.01.🚀",
  "query": """
    FROM library
    | EVAL release_month = DATE_TRUNC(1 month, release_date)
    | KEEP release_month
    | SORT release_month ASC
    | LIMIT 3
  """
}
----
// TEST[setup:library]

will remain valid, even if a future version of {esql} introduces
syntax changes or changes how the used commands or functions work.

> A released {esql} version will not receive breaking changes and
> will remain supported until it is deprecated.
> Already released {esql} versions will still profit from new features,
> bug fixes and performance improvements as long as those do not
> require breaking changes.

Reviewer: ```suggestion
We won't make breaking changes to released {esql} versions and
versions will remain supported until they are deprecated.
New features, bug fixes, and performance improvements 
will be continue to be added to released {esql} versions, 
provided they do not involve breaking changes.
```

Reviewer: rewording for clarity, apologies if suggestion messes up line lengths 🙈
